docs/design/yarn-v1.md

```diff
@@ -0,0 +1,169 @@
+# Cachi2 Yarn v1 (Yarn Classic) Prefetch Design
+
+## Yarn Overview
+[Yarn](https://github.com/yarnpkg/yarn) is a package manager for JavaScript. While Yarn v1.x is no longer actively developed and has been
+succeeded by [Yarn v2+ (Berry)](https://github.com/yarnpkg/berry), it [remains widely used](https://npmtrends.com/yarn).
+
+## Cachito Implementation Overview
+Cachito supports the Yarn 1.x package manager. It directly parses package.json files and uses our [pyarn](https://github.com/containerbuildsystem/pyarn)
```

Comment:
I'd probably refrain from mentioning Cachito too much. It is okay to mention that the core logic will be ported from it at the beginning of the doc, but then approach the doc like a completely new thing to cachi2, i.e. doing an overview of the functionality, mention the pitfalls and how we intend to approach it and treat it basically as if cachito never existed.

Reply:
See also discussion in "cachi2 rubygems / bundler design" https://github.com/containerbuildsystem/cachi2/pull/565#pullrequestreview-2248406464

Reply:
Still disagree, that thread IMO still hasn't provided any legitimate argument on why it should be a good practice over just giving credit/mentioning it once.

Reply:
I dropped this section and just added a quick statement to the "References" section at the bottom, like with the Ruby design doc